client/state/initial-state.js

```diff
@@ -0,0 +1,66 @@
+/**
+ * External dependencies
+ */
+import debugModule from 'debug';
+import localforage from 'localforage';
+import pick from 'lodash/object/pick';
+
+/**
+ * Internal dependencies
+ */
+import { createReduxStore } from 'state';
+import { pruneSites } from 'state/sites/reducer';
+import { pruneThemes } from 'lib/themes/reducers';
+
+/**
+ * Module variables
+ */
+const debug = debugModule( 'calypso:state' );
+
+const localforageConfig = {
+	name: 'calypso',
+	storeName: 'calypso_store',
+	description: 'Calypso Storage'
+};
+function pruneState( state ) {
```

Comment:
perhaps, though "validate" seems to imply "won't mutate, just verify" to me. "prune" says "make smaller, without changing the overall structure", though the last clause is pretty weak. 

We're using it two ways. When saving, it's definitely a prune type operation. On load, it's more of a migration. 

maybe think about this more as serialization? have serialize/unserialize or save/load pairs?